Python/ast.c

```diff
@@ -40,6 +40,15 @@ static int validate_typeparam(type_param_ty);
                      node->col_offset, node->end_col_offset, node->lineno, node->end_lineno); \
         return 0; \
     } \
+    if (node->lineno < -1 || node->end_lineno < -1 || \
```

Comment:
If we keep allowing -1 I don't see much point in disallowing other values. It's a potential compatibility break for users passing in other invalid values, so we may be better off just making any negative value work.